### PR TITLE
Add msr_text_compression

### PR DESCRIPTION
**Pull request by jeromeku**
Add [MSR Abstractive Text Compression Dataset](https://msropendata.com/datasets/f8ce2ec9-7fbd-48f7-a8bb-2d2279373563)

### CODE REVIEW

**Comment by lhoestq** [comment]
the `RemoteDatasetTest ` error in the CI is fixed on master so it's fine

File: datasets/msr_text_compression/README.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
+---
+YAML tags:
```
Comment (lhoestq) [inline datasets/msr_text_compression/README.md:2]:
```suggestion
```

File: datasets/msr_text_compression/msr_text_compression.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    @property
+    def manual_download_instructions(self):
+        return """\
+  You should download the dataset from https://www.microsoft.com/en-us/download/details.aspx?id=54262
+  The webpage requires registration.
+  Unzip and please put the files from the extracted RawData folder under the following names
+  train.tsv, valid.tsv, and test.tsv in a dir of your choice,
+  which will be used as a manual_dir, e.g. `~/.manual_dir/msr_text_compression`
+  The data can then be loaded via:
+  `datasets.load_dataset("msr_text_compression", data_dir="~/.manual_dir/msr_text_compression")`.
```
Comment (lhoestq) [inline datasets/msr_text_compression/msr_text_compression.py:64]: clear instructions, thanks :)

File: datasets/msr_text_compression/README.md [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+    "ratings": {
+      "feature_type": "Sequence",
+      "feature": {
+        "feature_type": "Value",
+        "dtype": "int64"
+      }
+    }
+  }
+}
+```
```
Comment (lhoestq) [inline datasets/msr_text_compression/README.md:131]: here we expect descriptions of the different fields in natural language

could you change that please ?

**Review by lhoestq — COMMENTED** [review]
Really cool thanks !

I left a few comments

File: datasets/msr_text_compression/README.md (hunk 3 above)
Comment (jeromeku) [reply datasets/msr_text_compression/README.md]: fixed

File: datasets/msr_text_compression/README.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+| Description       | Source | Target | Average CPS | Meaning Quality | Grammar Quality |
+| :------------- | :----------: | -----------: | -----------: | -----------: | -----------: |
+|  1-Sentence | 3764   | 15523    | 4.12 | 2.78 | 2.81 |
+|  2-Sentence | 2405   | 10900    | 4.53 | 2.78 | 2.83 |
+
+**Note**: Average CPS = Average Compressions per Source Text
+
+### Data Fields
+```json
```
Comment (lhoestq) [inline datasets/msr_text_compression/README.md:94]:
````suggestion

```
````

**Review by lhoestq — APPROVED** [review]
Thanks !

**Pull request merged**